File: Back-ASPNET/src/RecruitmentAuth.Domain/Entities/QuestionAnswer.cs
namespace RecruitmentAuth.Domain.Entities;

public class QuestionAnswer
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string SubmissionId { get; set; } = null!;
    public string QuestionId { get; set; } = null!;
    public string? AnswerText { get; set; }
    public string? SelectedOptions { get; set; } // JSON string
    public bool? IsCorrect { get; set; }
    public decimal? PointsAwarded { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public virtual TestSubmission Submission { get; set; } = null!;
    public virtual Question Question { get; set; } = null!;
}

File: Back-laravel/app/Models/QuestionAnswer.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Concerns\HasUuids;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class QuestionAnswer extends Model
{
    use HasFactory, HasUuids;

    protected $fillable = [
        'submission_id',
        'question_id',
        'answer_text',
        'selected_options',
        'is_correct',
        'points_awarded',
    ];

    protected $casts = [
        'selected_options' => 'array',
        'is_correct' => 'boolean',
    ];

    public function submission()
    {
        return $this->belongsTo(TestSubmission::class, 'submission_id');
    }

    public function question()
    {
        return $this->belongsTo(Question::class);
    }
}
